### PR TITLE
Adding NextJS as an e2e testing point / app

### PR DESCRIPTION
**Pull request by frehner**
### Description

Adding NextJS as an e2e testing point / app. Getting the configs all setup so that if you run `yarn dev` at the root of the monorepo, it runs the watch script for hydrogen-react and also NextJS. Clarify that you can run the stories in just the hydrogen-react repo by running `yarn dev:story`

Note that we don't deploy any previews yet, but that's in the plans. 

### Additional context

---

### Before submitting the PR, please make sure you do the following:

- [x] Read the [Contributing Guidelines](https://github.com/shopify/hydrogen-ui/blob/main/contributing.md)
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`)
- [x] Update docs in this repository according to your change
- [ ] Run `yarn changeset add` if this PR cause a version bump based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/). If you have a breaking change, it will need to wait until the next major version release. Otherwise, use patch updates even for new features. Read [more about Hydrogen-UI's versioning.](https://github.com/shopify/hydrogen-ui/blob/main/readme.md#versioning)

### CODE REVIEW

File: apps/nextjs/gql/fragment-masking.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export function useFragment<TType>(
+  _documentNode: DocumentNode<TType, any>,
+  fragmentType: ReadonlyArray<FragmentType<DocumentNode<TType, any>>> | null | undefined
+): ReadonlyArray<TType> | null | undefined
+export function useFragment<TType>(
+  _documentNode: DocumentNode<TType, any>,
+  fragmentType: FragmentType<DocumentNode<TType, any>> | ReadonlyArray<FragmentType<DocumentNode<TType, any>>> | null | undefined
+): TType | ReadonlyArray<TType> | null | undefined {
+  return fragmentType as any
+}
```
Comment (lordofthecactus) [inline apps/nextjs/gql/fragment-masking.ts:40]: Is this all generated by nextjs? Is this considered common knowledge? Maybe some comments for people who don't know what this is.

File: apps/nextjs/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  try {
+    const response = await request({
+      url: shopClient.getStorefrontApiUrl(),
+      document: query,
+      // @TODO: convert to 'getPrivateTokenHeaders({buyerIp})'
+      requestHeaders: shopClient.getPublicTokenHeaders(),
+    });
+
+    // @TODO I don't love how we do this with 'errors' and 'data'
+    return {props: {data: response, errors: null}};
```
Comment (lordofthecactus) [inline apps/nextjs/pages/index.tsx:30]: seems to be a pattern in the community to do this? like:
```
const {data, errors} = useQuery(...)
```

I'm ok with it

File: apps/nextjs/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      <Head>
+        <title>Create Next App</title>
+        <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <main className={styles.main}>
+        <h1>Welcome to {data?.shop.name} on NextJS</h1>
+
+        {/* @TODO Using hydrogen-react's <Image/> is nice, but we should also provide our 'loader' so you can used NextJS' Image component as well */}
```
Comment (lordofthecactus) [inline apps/nextjs/pages/index.tsx:56]: 💯

**Review by lordofthecactus — COMMENTED** [review]
Overall the app looks good as a starting point 💯 

I'm wondering what are the plans with this? is there an issue tracking it?

File: apps/nextjs/gql/fragment-masking.ts (hunk 1 above)
Comment (frehner) [reply apps/nextjs/gql/fragment-masking.ts:40]: Yeah, sorry I should have left a comment in the PR. Anything in the `apps/nextjs/gql` folder is auto-generated by the tooling

**Pull request merged**